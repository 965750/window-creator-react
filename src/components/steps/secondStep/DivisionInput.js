import React from 'react'
import styles from './DivisionInput.module.scss'
import PropTypes from 'prop-types'

const DivisionInput = ({ changeDivision, id, inputValue, classes }) => {
    return (
        <div className={`${classes ? classes : ''} flex`}>
            <div
                className={`${styles.inputBox} border border-gullGray text-center`}
            >
                {inputValue}
            </div>
            <button
                onClick={() => changeDivision(1, id)}
                className={`${styles.inputBox} ml-1 bg-botticelli text-center cursor-pointer`}
            >
                +
            </button>
            <button
                onClick={() => changeDivision(-1, id)}
                className={`${styles.inputBox} ml-1 bg-botticelli text-center cursor-pointer`}
            >
                -
            </button>
        </div>
    )
}

DivisionInput.propTypes = {
    changeDivision: PropTypes.func,
    id: PropTypes.string,
    inputValue: PropTypes.number,
    classes: PropTypes.string,
}

export default DivisionInput
