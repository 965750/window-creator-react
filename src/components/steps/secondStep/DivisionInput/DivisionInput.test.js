import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import DivisionInput from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('DivisionInput', () => {
  let wrapper
  const initialProps = {
    id: 'mockId',
    inputValue: 7,
    changeDivision: jest.fn(),
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <DivisionInput {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should pass classes from props to wrapper', () => {
    const mockClasses = 'mockClasses'

    setUp({
      ...initialProps,
      classes: mockClasses,
    })

    expect(wrapper.queryByTestId('divisionInputWrapper')).toHaveClass(mockClasses)
  })
})
