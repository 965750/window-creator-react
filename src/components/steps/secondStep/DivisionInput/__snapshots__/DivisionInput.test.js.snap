// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DivisionInput should match snapshot 1`] = `
<DocumentFragment>
  <div
    class=" flex"
    data-testid="divisionInputWrapper"
  >
    <div
      class="inputBox border border-gullGray text-center"
    >
      7
    </div>
    <button
      class="inputBox ml-1 bg-botticelli text-center cursor-pointer"
    >
      +
    </button>
    <button
      class="inputBox ml-1 bg-botticelli text-center cursor-pointer"
    >
      -
    </button>
  </div>
</DocumentFragment>
`;
